test/test_mps.py

```diff
@@ -4288,7 +4291,383 @@ def test_legacy_constructor(self):
+        'split_with_sizes': None,
+        'trapezoid': None,
+        'eq': None,
+        'mul': None,
+        'cartesian_prod': None,
+        'nonzero': None,
+        'bool': None,
+    }
+
+    # Used for accept mode only
+    NEW_ALLOW_LIST = defaultdict(list)
+
+    @ops(op_db)
+    def test_output_match(self, device, dtype, op):
+        self.assertEqual(device, "cpu")
+        if not torch.backends.mps.is_available():
+            self.skipTest("MPS is not available")
+
+        if dtype in [torch.bfloat16, torch.double, torch.int8]:
+            self.skipTest(f"{dtype} not supported by MPS")
+
+        if dtype.is_complex:
+            self.skipTest("The MPS backend does not support complex types")
+
+        key = op.name + op.variant_test_name
+        if key in self.BLOCKLIST:
+            if self.BLOCKLIST[key] is None or dtype in self.BLOCKLIST[key]:
+                self.skipTest(f"Running test with {op.name} hangs so skipping")
+
+        # Make this an expecttest manually
+        # When this env variable is set, generate a new ALLOWLIST_OP
+        # that reflects the current state of what passes or not
+        if os.environ.get("EXPECTTEST_ACCEPT", None) == "1":
+            generate_new_truth = True
+        else:
+            generate_new_truth = False
+
+        if not generate_new_truth:
+            if op.name not in self.ALLOWLIST_OP:
+                self.skipTest(f"{op.name} is not in the allow list for test on MPS")
+            else:
+                if str(dtype) not in self.ALLOWLIST_OP[op.name]:
+                    self.skipTest(f"{op.name} is in the allow list for MPS but {dtype} is excluded")
+
+        try:
+            cpu_samples = op.sample_inputs(device, dtype)
+
+            for cpu_sample in cpu_samples:
+                cpu_args = [cpu_sample.input] + list(cpu_sample.args)
+                cpu_kwargs = cpu_sample.kwargs
+                mps_args = tuple(arg.to("mps") if isinstance(arg, torch.Tensor) else arg for arg in cpu_args)
+                mps_kwargs = {k: v.to("mps") if isinstance(v, torch.Tensor) else v for k, v in cpu_kwargs.items()}
+
+                if True or op.supports_mps:
+                    cpu_out = op(*cpu_args, **cpu_kwargs)
+                    # if op.name == "repeat":
+                    #     print(cpu_args)
+                    #     print(cpu_out)
+                    mps_out = op(*mps_args, **mps_kwargs)
+                    self.assertEqual(cpu_out, mps_out)
+                else:
+                    with self.assertRaisesRegex(NotImplementedError, "not impls?"):
+                        mps_out = op(*cpu_args, **mps_kwargs)
+        except Exception as e:
+            if not generate_new_truth:
+                raise e
+        else:
+            if generate_new_truth:
+                self.NEW_ALLOW_LIST[op.name].append(str(dtype))
+
+        # We could write it only once. But I don't know how to detect that the current test is the last one
```

Comment:
Holy cow though -- can't this at least condition on whether the list was changed or not by the above?

Reply:
Sure.
Note that this is only ever run locally. And this whole code should go away in a month once we can just integrate with OpInfo properly.